ykman-gui/py/yubikey.py

```diff
@@ -578,6 +579,59 @@ def piv_unblock_pin(self, puk, new_pin):
                     'message': str(e),
                 }
 
+    def piv_can_parse(self, file_url):
+        try:
+            file_path = urllib.parse.urlparse(file_url).path
+            with open(file_path, 'r+b') as file:
+                data = file.read()
+                try:
+                    parse_certificate(data, password=None)
+                    return {'success': True, 'error': None}
+                except (ValueError, TypeError):
+                    pass
+                try:
+                    parse_private_key(data, password=None)
+                    return {'success': True, 'error': None}
+                except (ValueError, TypeError):
+                    pass
+            raise ValueError('Failed to parse certificate or key')
+        except Exception as e:
+            return {'success': False, 'error': str(e)}
```

Comment:
Suggestion: 
```
return {'success': False, 'error': type(e).__name__, 'message': str(e)}
```
to make this integrate nicely with #90. Leaving the `error` out and specifying only `message` works too.